app/controllers/helpers.rb

```diff
@@ -0,0 +1,55 @@
+module Helpers
+	def self.available
+		true
+	end
+
+	def self.can_edit?(session, tweet)
+		#binding.pry
+		if session[:id]
+			tweet.user.id==User.find(session[:id]).id
+		else
+			false
+		end
+	end
+
+	def self.modify_tweet(tweet, params)
+		#binding.pry
+		tweet.content=params[:content]
+		tweet.save
+
+	end
+
+	def self.valid_login(params)
+		!!User.find_by(username: params[:username], password: params[:password])
```

Comment:
It's better to rely on the successful call to `current_user`. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```